app/gosqs/delete_message_batch.go

```diff
@@ -0,0 +1,114 @@
+package gosqs
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/Admiral-Piett/goaws/app"
+	"github.com/Admiral-Piett/goaws/app/interfaces"
+	"github.com/Admiral-Piett/goaws/app/models"
+	"github.com/Admiral-Piett/goaws/app/utils"
+	"github.com/gorilla/mux"
+	log "github.com/sirupsen/logrus"
+)
+
+func DeleteMessageBatchV1(req *http.Request) (int, interfaces.AbstractResponseBody) {
+
+	requestBody := models.NewDeleteMessageBatchRequest()
+	ok := utils.REQUEST_TRANSFORMER(requestBody, req, false)
+
+	if !ok {
+		log.Error("Invalid Request - DeleteMessageBatchV1")
+		return utils.CreateErrorResponseV1("InvalidParameterValue", true)
+	}
+
+	queueUrl := requestBody.QueueUrl
+
+	queueName := ""
+	if queueUrl == "" {
+		vars := mux.Vars(req)
+		queueName = vars["queueName"]
+	} else {
+		uriSegments := strings.Split(queueUrl, "/")
+		queueName = uriSegments[len(uriSegments)-1]
+	}
+
+	if _, ok := app.SyncQueues.Queues[queueName]; !ok {
+		return utils.CreateErrorResponseV1("QueueNotFound", true)
+	}
+
+	if len(requestBody.Entries) == 0 {
+		return utils.CreateErrorResponseV1("EmptyBatchRequest", true)
+	}
+
+	if len(requestBody.Entries) > 10 {
+		return utils.CreateErrorResponseV1("TooManyEntriesInBatchRequest", true)
+	}
+
+	ids := map[string]struct{}{}
+	for _, v := range requestBody.Entries {
+		if _, ok := ids[v.Id]; ok {
+			return utils.CreateErrorResponseV1("BatchEntryIdsNotDistinct", true)
+		}
+		ids[v.Id] = struct{}{}
+	}
+
+	deleteEntries := make([]deleteEntry, len(requestBody.Entries))
+	for i, entry := range requestBody.Entries {
+		deleteEntries[i] = deleteEntry{
+			Id:            entry.Id,
+			ReceiptHandle: entry.ReceiptHandle,
+			Deleted:       false,
+		}
+	}
+
+	app.SyncQueues.Lock()
+	defer app.SyncQueues.Unlock()
+
+	deletedEntries := make([]models.DeleteMessageBatchResultEntry, 0)
+	for j, deleteEntry := range deleteEntries {
+		for i, msg := range app.SyncQueues.Queues[queueName].Messages {
```

Comment:
I know this was the existing code, but this pattern feels labor intensive.  If nothing else I think it might be more performant to 1. fetch out the queue once, and 2. at least flip the order of these loops if not use a data structure that doesn't need the 2 loops, like a map.  If we just flipped the order, we'd know that we only have a max of 10 entries (unless we remove that requirement above), in `deleteEntries`.  In the messages, in theory, we could have an unlimited amount.

You don't have to change it now, but would you mind putting `# TODO - Admiral-Piett` with a link to this comment in there, so I can come back to it at some point?

Reply:
@Admiral-Piett
Our friend, Mr. Otsuka(you can see the name on the commit 😁 ) did the refactoring!

- Create a Map called deleteMessageMap in advance, with ReceiptHandle as the key and deleteEntry as the value.
- Scan the messages in the Queue and match them with the ReceiptHandle of the deleteEntry to determine which messages to delete.
- If the ReceiptHandle matches, determine that the message is to be deleted and perform the delete process as before.

Reply:
Awesome!  Thank you both then!  That looks great!  I really appreciate it!